src/otp-field.tsx

```diff
@@ -0,0 +1,309 @@
+import {useFieldProps} from './form-context';
+import {createChangeEvent} from './utils/dom';
+import {HelperText} from './text-field-components';
+import {flushSync} from 'react-dom';
+
+// Protection for when there is more than one OtpField in the page.
+// This should't be a supported use case, but we need it in storybook/playroom, and for some reason
+// some Chrome versions crash when using navigator.credentials.get() more than once simultaneously.
+let isWaitingForSms = false;
+
+type OtpInputProps = {
+    length?: number;
+    hideCode?: boolean;
+    disabled?: boolean;
+    readOnly?: boolean;
+    value?: string;
+    defaultValue?: string;
+    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    inputRef: (field: HTMLInputElement | null) => void;
+};
+
+const OtpInput = ({
+    length = 6,
+    hideCode = false,
+    disabled,
+    readOnly,
+    value,
+    defaultValue,
+    onChange,
+    inputRef,
+}: OtpInputProps): React.ReactElement => {
+    const {texts} = useTheme();
+    const [selfValue, setSelfValue] = React.useState<string>(defaultValue?.slice(0, length) ?? '');
+    const [focusIndex, setFocusIndex] = React.useState<number | undefined>(undefined);
+
+    const inputsList: Array<HTMLInputElement | null> = React.useRef(Array.from({length}, () => null)).current;
+
+    const isControlledByParent = typeof value !== 'undefined';
+    const controlledValue: string = isControlledByParent ? value.slice(0, length) : selfValue;
+
+    const changeValue = React.useCallback(
+        (newValue: string) => {
+            if (newValue === controlledValue) {
+                return;
+            }
+            if (!isControlledByParent) {
+                setSelfValue(newValue);
+            }
+            const firstInput = inputsList[0];
+            if (firstInput) {
+                onChange?.(createChangeEvent(firstInput, newValue));
+            }
+        },
+        [controlledValue, inputsList, isControlledByParent, onChange]
+    );
+
+    React.useEffect(() => {
+        // https://developer.mozilla.org/en-US/docs/Web/API/WebOTP_API
+        if ('OTPCredential' in window && !isWaitingForSms) {
+            isWaitingForSms = true;
+            const abortController = new AbortController();
+            navigator.credentials
+                .get({
+                    // @ts-expect-error: otp is not in the types yet
+                    otp: {transport: ['sms']},
+                    signal: abortController.signal,
+                })
+                .then((otp) => {
+                    if (otp) {
+                        // @ts-expect-error: otp is not in the types yet
+                        const code = otp.code.slice(0, length);
+                        changeValue(code);
+                    }
+                })
+                .catch(() => {
+                    // ignore;
+                });
+            return () => {
+                isWaitingForSms = false;
```

Comment:
this clears the lock when the component unmounts. Perhaps it would be enough to do it on the `credentials` promise `.finally()`

Reply:
hmm... no because I'm aborting it on unmount too. But I could unlock in unmount and in .finally()